modules/stochastic_tools/include/utils/GaussianProcess.h

```diff
@@ -119,64 +125,33 @@ class GaussianProcessHandler
    */
   void standardizeData(RealEigenMatrix & data, bool keep_moments = false);
 
-  /**
-   * Tune the hyper parameters in the covariance function using PETSc-TAO.
-   * @param training_params The training parameter values (x values) for the
-   *                        covariance matrix.
-   * @param training_data The training data (y values) for the inversion of the
-   *                      covariance matrix.
-   * @param tao_options Additional options for TAO.
-   * @param show_optimization_details Switch to show details of TAO or Adam optimization.
-   */
-  PetscErrorCode tuneHyperParamsTAO(const RealEigenMatrix & training_params,
-                                    const RealEigenMatrix & training_data,
-                                    std::string tao_options = "",
-                                    bool verbose = false);
-
-  /// Used to form initial guesses in the TAO optimization routines
-  PetscErrorCode formInitialGuessTAO(Vec theta_vec);
-
-  /// Build the bounds for the hyper parameter optimization with TAO
-  void buildHyperParamBoundsTAO(libMesh::PetscVector<Number> & theta_l,
-                                libMesh::PetscVector<Number> & theta_u) const;
-
-  // Wrapper for PETSc function callback
-  static PetscErrorCode
-  formFunctionGradientWrapper(Tao tao, Vec theta, PetscReal * f, Vec Grad, void * ptr);
-
-  // Computes Gradient of the loss function for TAO usage
-  void formFunctionGradient(Tao tao, Vec theta, PetscReal * f, Vec Grad);
-
   // Tune hyperparameters using Adam
   void tuneHyperParamsAdam(const RealEigenMatrix & training_params,
                            const RealEigenMatrix & training_data,
-                           unsigned int iter,
-                           const unsigned int & batch_size,
-                           const Real & learning_rate,
-                           const bool & verbose);
+                           const GPOptimizerOptions & opts);
 
-  // Computes the loss function for Adam usage
-  Real getLossAdam(RealEigenMatrix & inputs, RealEigenMatrix & outputs);
+  // Computes the loss function
+  Real getLoss(RealEigenMatrix & inputs, RealEigenMatrix & outputs);
 
-  // Computes Gradient of the loss function for Adam usage
-  std::vector<Real> getGradientAdam(RealEigenMatrix & inputs);
+  // Computes Gradient of the loss function
+  std::vector<Real> getGradient(RealEigenMatrix & inputs);
```

Comment:
I couldn't make the `getLoss` const because it modifies member variables to cache the results of expensive operations. I made everything else const that could be const.